test/e2e/chaos/basic.go

```diff
@@ -1640,151 +1621,210 @@ selector:
+				framework.ExpectEqual(len(allSlowConnection(c, networkPeers, ports)), 0)
+
+				complicateNetem := makeNetworkDelayChaos(
+					ns, "network-chaos-3",
+					map[string]string{"app": "network-peer-0"},
+					map[string]string{},
+					v1alpha1.OnePodMode,
+					v1alpha1.OnePodMode,
+					testDelayTcParamEvenMoreComplicate,
+					testDelayDuration,
+					testDelaySchedulerSpec,
+				)
 
-				complicateNetem := &v1alpha1.NetworkChaos{
-					ObjectMeta: metav1.ObjectMeta{
-						Name:      "network-chaos-3",
-						Namespace: ns,
-					},
-					Spec: v1alpha1.NetworkChaosSpec{
-						Action: v1alpha1.DelayAction,
-						Selector: v1alpha1.SelectorSpec{
-							Namespaces:     []string{ns},
-							LabelSelectors: map[string]string{"app": "network-peer-0"},
-						},
-						Mode: v1alpha1.OnePodMode,
-						TcParameter: v1alpha1.TcParameter{
-							Delay: &v1alpha1.DelaySpec{
-								Latency:     "200ms",
-								Correlation: "25",
-								Jitter:      "0ms",
-							},
-							Loss: &v1alpha1.LossSpec{
-								Loss:        "25",
-								Correlation: "25",
-							},
-							Duplicate: &v1alpha1.DuplicateSpec{
-								Duplicate:   "25",
-								Correlation: "25",
-							},
-							Corrupt: &v1alpha1.CorruptSpec{
-								Corrupt:     "25",
-								Correlation: "25",
-							},
-						},
-						Duration: pointer.StringPtr("9m"),
-						Scheduler: &v1alpha1.SchedulerSpec{
-							Cron: "@every 10m",
-						},
-					},
-				}
 				klog.Infof("Injecting delay for 0")
 				err = cli.Create(ctx, complicateNetem.DeepCopy())
 				framework.ExpectNoError(err, "create network chaos error")
 				time.Sleep(5 * time.Second)
-				framework.ExpectEqual(allSlowConnection(), [][]int{{0, 1}, {0, 2}, {0, 3}})
+				framework.ExpectEqual(allSlowConnection(c, networkPeers, ports), [][]int{{0, 1}, {0, 2}, {0, 3}})
+
+				cancel()
+			})
+		})
+
+		ginkgo.Context("[NetworkPartitionWithNetem]", func() {
+			ginkgo.It("[MixSchedule]", func() {
+				ctx, cancel := context.WithCancel(context.Background())
+
+				for index := range networkPeers {
+					err = waitE2EHelperReady(c, ports[index])
+
+					framework.ExpectNoError(err, "wait e2e helper ready error")
+				}
+
+				framework.ExpectEqual(len(allBlockedConnection(c, networkPeers, ports)), 0)
+				framework.ExpectEqual(len(allSlowConnection(c, networkPeers, ports)), 0)
+
+				var (
+					testDelayTcParam = v1alpha1.TcParameter{
+						Delay: &v1alpha1.DelaySpec{
+							Latency:     "200ms",
+							Correlation: "25",
+							Jitter:      "0ms",
+						},
+					}
+					testDelayDuration      = pointer.StringPtr("9m")
+					testDelaySchedulerSpec = &v1alpha1.SchedulerSpec{
+						Cron: "@every 10m",
+					}
+				)
+
+				networkPartition := makeNetworkPartitionChaos(
+					ns, "network-chaos-1",
+					map[string]string{"app": "network-peer-0"},
+					map[string]string{"app": "network-peer-1"},
+					v1alpha1.OnePodMode,
+					v1alpha1.OnePodMode,
+					v1alpha1.To,
```

Comment:
It will only inject iptables rules on `network-peer-0`

Reply:
Okay, I changed to 2->0, so now peer-2 partitioned with peer-0, and delayed with peer-1.